{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "20f6721b",
   "metadata": {},
   "outputs": [],
   "source": [
    "options(warn=-1)\n",
    "suppressMessages({library(\"dplyr\")\n",
    "library(\"tidyr\")\n",
    "library(\"ggplot2\")\n",
    "library(\"GenomicRanges\")\n",
    "library(\"readxl\")\n",
    "library(\"data.table\")\n",
    "library(\"rtracklayer\")\n",
    "library(\"PRROC\")})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c3c2cdc",
   "metadata": {},
   "source": [
    "# Combining and tydying CRISPRi data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "af79eea3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining, by = \"gene_short_name\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overall we analyzed 362 CRISPRi-verified E-P pairs in K562.\n",
      "219 (60.5%) of them had an underlying (physical) contact in MChIP-C data,\n",
      "70 (19.3%) - in Micro-C data,\n",
      "7 (1.9%) - in Hi-C data.\n",
      "As a control we also analyzed physical contacts within 54930 non-functional DHS-P pairs.\n",
      "3521 (6.4%) of them had an underlying (physical) contact in MChIP-C data,\n",
      "915 (1.7%) - in Micro-C data,\n",
      "321 (0.6%) - in Hi-C data.\n",
      "For PLAC-seq we were able to analyze only 311 verified E-P pairs and 54600 non-functional DHS-P pairs\n",
      "due to the lower resolution of PLAC-seq interaction calling data.\n",
      "43 (13.8%) of analyzed E-P pairs and 1031 (1.9%) of analyzed non-functional DHS-P pairs\n",
      "had an underlying (physical) PLAC-seq contact.\n",
      "We also calculated precision metrics for E-P predictors based on studied proximity-ligation methods.\n",
      "Precisions were estimated to be equal to: 5.9% (MChIP-C),4% (PLAC-seq),7.1% (Micro-C) and 2.1% (Hi-C).\n",
      "Finally we analyzed 50%-, 25%- and 10%-subsampled MChIP-C datasets. Their sensitivity was (correspondingly): 56.1%, 45%, and 18.2%. False positive rate: 5.5%, 3.1%, and 1%. Precision: 6.3%, 8.7%, and 10.4%."
     ]
    }
   ],
   "source": [
    "overlapPairsWithLoops=function(loop_file, shift=0){\n",
    "    loops <- read.csv(loop_file, header = F, sep = \"\\t\")\n",
    "    loop_vp_ranges <- GRanges(seqnames = loops$V1, ranges=IRanges(start=loops$V2, end=loops$V3))\n",
    "    loop_OE_ranges <- GRanges(seqnames = loops$V4, ranges=IRanges(start=loops$V5-shift, end=loops$V6+shift))\n",
    "    output <- CRISPRi$id %in% as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_OE_ranges))[paste(as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_OE_ranges))$queryHits,\n",
    "        as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_OE_ranges))$subjectHits) %in% paste(as.data.frame(findOverlaps(CRISPRi_TSS_ranges, loop_vp_ranges))$queryHits, \n",
    "        as.data.frame(findOverlaps(CRISPRi_TSS_ranges, loop_vp_ranges))$subjectHits),]$queryHits\n",
    "    return(output)}\n",
    "overlapPairsWithLoopsSym=function(loop_file, shift=2500){\n",
    "    loops <- read.csv(loop_file, header = F, sep = \"\\t\")\n",
    "    loop_bin1_ranges <- GRanges(seqnames = loops$V1, ranges=IRanges(start=loops$V2-shift, end=loops$V3+shift))\n",
    "    loop_bin2_ranges <- GRanges(seqnames = loops$V4, ranges=IRanges(start=loops$V5-shift, end=loops$V6+shift))\n",
    "    output <- CRISPRi$id %in% unique(as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_bin2_ranges))[paste(as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_bin2_ranges))$queryHits,\n",
    "        as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_bin2_ranges))$subjectHits) %in% paste(as.data.frame(findOverlaps(CRISPRi_TSS_ranges, loop_bin1_ranges))$queryHits, \n",
    "        as.data.frame(findOverlaps(CRISPRi_TSS_ranges, loop_bin1_ranges))$subjectHits),]$queryHits) | \n",
    "        CRISPRi$id %in% unique(as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_bin1_ranges))[paste(as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_bin1_ranges))$queryHits, \n",
    "        as.data.frame(findOverlaps(CRISPRi_OE_ranges, loop_bin1_ranges))$subjectHits) %in% paste(as.data.frame(findOverlaps(CRISPRi_TSS_ranges, loop_bin2_ranges))$queryHits, \n",
    "        as.data.frame(findOverlaps(CRISPRi_TSS_ranges, loop_bin2_ranges))$subjectHits),]$queryHits)\n",
    "    return(output)}\n",
    "\n",
    "gasperini_neg <- read.csv(\"Auxiliary_data/CRISPRi_data/GSE120861_all_deg_results.at_scale.txt.gz\", \n",
    "                          header = T, sep = \"\\t\")\n",
    "genes <- unique(gasperini_neg[,c(9,16)])\n",
    "genes <- filter(genes, !(duplicated(genes$gene_short_name)))\n",
    "gasperini_neg <- filter(gasperini_neg, quality_rank_grna==\"top_two\" & !(outlier_gene) &\n",
    "                        fold_change.transcript_remaining<1.1 &\n",
    "                        (pvalue.empirical.adjusted > 0.1|pvalue.empirical.adjusted==\"not_applicable\"))\n",
    "gasperini_neg <- gasperini_neg[,c(12:14,9,16)]\n",
    "gasperini_neg$Significant <- FALSE\n",
    "gasperini_neg[,2] <- as.integer(gasperini_neg[,2])\n",
    "gasperini_neg[,3] <- as.integer(gasperini_neg[,3])\n",
    "gasperini_pos <- read_excel(\"Auxiliary_data/CRISPRi_data/Gasperini2019_TableS2.xlsx\", sheet = 3)\n",
    "gasperini_pos <- filter(gasperini_pos, high_confidence_subset)\n",
    "gasperini_pos <- gasperini_pos[,c(9:11,3)]\n",
    "colnames(gasperini_pos) <- colnames(gasperini_neg)[1:4]\n",
    "gasperini_pos <- left_join(gasperini_pos,genes)\n",
    "gasperini_pos$Significant <- TRUE\n",
    "fulco <- read_excel(\"Auxiliary_data/CRISPRi_data/Fulco2019_SupTables.xlsx\", sheet = 16, skip = 1)\n",
    "fulco <- filter(fulco, class != \"promoter\" & \n",
    "                ((`Fraction change in gene expr`< 0 & Significant == T)| (Significant == F)))\n",
    "fulco <- fulco[,c(1,2,3,5,18,10)]\n",
    "colnames(fulco) <- colnames(gasperini_neg)\n",
    "CRISPRi <- bind_rows(gasperini_pos, gasperini_neg,fulco)\n",
    "CRISPRi<-filter(CRISPRi,abs((CRISPRi$target_site.start+CRISPRi$target_site.stop)/2-CRISPRi$target_gene.start)<1000000&\n",
    "                abs((CRISPRi$target_site.start+CRISPRi$target_site.stop)/2-CRISPRi$target_gene.start)>5000)\n",
    "CRISPRi_TSS_ranges <- GRanges(seqnames = CRISPRi$target_site.chr, \n",
    "                              ranges=IRanges(start=CRISPRi$target_gene.start, end=CRISPRi$target_gene.start+1))\n",
    "vp <- read.csv(\"MChIPC_output/MChIPC_viewpoints.bed\", header=F, sep=\"\\t\") \n",
    "# here I should later add coverage filter\n",
    "vp_ranges <- GRanges(seqnames = vp$V1, ranges=IRanges(start=vp$V2, end=vp$V3))\n",
    "CRISPRi <- CRISPRi[as.data.frame(findOverlaps(CRISPRi_TSS_ranges, vp_ranges))$queryHits,]\n",
    "CRISPRi_TSS_ranges <- GRanges(seqnames = CRISPRi$target_site.chr, \n",
    "                             ranges=IRanges(start=CRISPRi$target_gene.start, end=CRISPRi$target_gene.start+1))\n",
    "CRISPRi_OE_ranges <- GRanges(seqnames = CRISPRi$target_site.chr, \n",
    "                             ranges=IRanges(start=CRISPRi$target_site.start, end=CRISPRi$target_site.stop))\n",
    "CRISPRi$id <- seq(1,nrow(CRISPRi))\n",
    "CRISPRi$MChIPC.loop <- overlapPairsWithLoops(\"MChIPC_output/MChIPC_interactions.bedpe\", shift=500)\n",
    "CRISPRi$PLACseq.loop <- overlapPairsWithLoops(\"PLACseq_output/PLACseq_interactions.bedpe\")\n",
    "CRISPRi$MicroC.loop <- overlapPairsWithLoopsSym(\"MicroC_output/MicroC_interactions.bedpe\", shift=2500)\n",
    "CRISPRi$HiC.loop <- overlapPairsWithLoopsSym(\"HiC_output/HiC_interactions.bedpe\", shift=2500)\n",
    "CRISPRi$MChIPC.sub05.loop <- overlapPairsWithLoops(\"MChIPC_output/MChIPC_interactions.sub_0.5.bedpe\", shift=500)\n",
    "CRISPRi$MChIPC.sub025.loop <- overlapPairsWithLoops(\"MChIPC_output/MChIPC_interactions.sub_0.25.bedpe\", shift=500)\n",
    "CRISPRi$MChIPC.sub01.loop <- overlapPairsWithLoops(\"MChIPC_output/MChIPC_interactions.sub_0.1.bedpe\", shift=500)\n",
    "CTCF_peaks <- read.csv(\"Auxiliary_data/ENCFF002CEL.bed.gz\", header = F, sep = \"\\t\")\n",
    "CTCF_peaks <- filter(CTCF_peaks, V5 > 250)\n",
    "CTCF_ranges <- GRanges(seqnames = CTCF_peaks$V1, \n",
    "                       ranges=IRanges(start=CTCF_peaks$V2-1000, end=CTCF_peaks$V3+1000))\n",
    "CRISPRi$overlaps.CTCF <- CRISPRi$id %in% as.data.frame(findOverlaps(CRISPRi_OE_ranges, CTCF_ranges))$queryHits\n",
    "CRISPRi$id <- NULL\n",
    "CRISPRi_pos <- filter(CRISPRi, Significant)\n",
    "CRISPRi_neg <- filter(CRISPRi, !(Significant))\n",
    "# some basic statistics:\n",
    "cat(paste0(\"Overall we analyzed \",nrow(CRISPRi_pos),\" CRISPRi-verified E-P pairs in K562.\\n\"))\n",
    "cat(paste0(nrow(filter(CRISPRi_pos,MChIPC.loop)),\" (\",round(100*nrow(filter(CRISPRi_pos,MChIPC.loop))/nrow(CRISPRi_pos),1),\n",
    "\"%) of them had an underlying (physical) contact in MChIP-C data,\\n\"))\n",
    "cat(paste0(nrow(filter(CRISPRi_pos, MicroC.loop)),\" (\",round(100*nrow(filter(CRISPRi_pos,MicroC.loop))/nrow(CRISPRi_pos),1),\n",
    "\"%) - in Micro-C data,\\n\"))\n",
    "cat(paste0(nrow(filter(CRISPRi_pos, HiC.loop)),\" (\",round(100*nrow(filter(CRISPRi_pos,HiC.loop))/nrow(CRISPRi_pos),1),\n",
    "\"%) - in Hi-C data.\\n\"))\n",
    "#cat(paste0(\"When we considered only \",nrow(filter(CRISPRi_pos, coverage>5000)),\n",
    "#\" E-P pairs with high-coverage viewpoints we found out that even higher proportion (\",\n",
    "#100*nrow(filter(CRISPRi_pos, coverage>5000&MChIPC.loop))/nrow(filter(CRISPRi_pos, coverage>5000)),\n",
    "#\"%) of them had underlying (physical) contact in MChIP-C data.\"))\n",
    "      \n",
    "cat(paste0(\"As a control we also analyzed physical contacts within \",nrow(CRISPRi_neg),\" non-functional DHS-P pairs.\\n\"))\n",
    "cat(paste0(nrow(filter(CRISPRi_neg,MChIPC.loop)),\" (\",round(100*nrow(filter(CRISPRi_neg,MChIPC.loop))/nrow(CRISPRi_neg),1), \n",
    "\"%) of them had an underlying (physical) contact in MChIP-C data,\\n\"))\n",
    "cat(paste0(nrow(filter(CRISPRi_neg, MicroC.loop)),\" (\",round(100*nrow(filter(CRISPRi_neg,MicroC.loop))/nrow(CRISPRi_neg),1),\n",
    "\"%) - in Micro-C data,\\n\"))\n",
    "cat(paste0(nrow(filter(CRISPRi_neg, HiC.loop)),\" (\",round(100*nrow(filter(CRISPRi_neg,HiC.loop))/nrow(CRISPRi_neg),1),\n",
    "\"%) - in Hi-C data.\\n\"))\n",
    "\n",
    "N_PLAC_pos <- nrow(filter(CRISPRi_pos,abs((CRISPRi_pos$target_site.start+CRISPRi_pos$target_site.stop)/2-CRISPRi_pos$target_gene.start)>10000))\n",
    "N_PLAC_neg <- nrow(filter(CRISPRi_neg,abs((CRISPRi_neg$target_site.start+CRISPRi_neg$target_site.stop)/2-CRISPRi_neg$target_gene.start)>10000))\n",
    "cat(paste0(\"For PLAC-seq we were able to analyze only \",N_PLAC_pos,\" verified E-P pairs and \", N_PLAC_neg,\n",
    "\" non-functional DHS-P pairs\\ndue to the lower resolution of PLAC-seq interaction calling data.\\n\"))            \n",
    "cat(paste0(nrow(filter(CRISPRi_pos, PLACseq.loop)),\" (\",round(100*nrow(filter(CRISPRi_pos,PLACseq.loop))/N_PLAC_pos,1), \n",
    "\"%) of analyzed E-P pairs and \",\n",
    "nrow(filter(CRISPRi_neg, PLACseq.loop)),\" (\",round(100*nrow(filter(CRISPRi_neg,PLACseq.loop))/N_PLAC_neg,1),\n",
    "\"%) of analyzed non-functional DHS-P pairs\\nhad an underlying (physical) PLAC-seq contact.\\n\"))\n",
    "cat(\"We also calculated precision metrics for E-P predictors based on studied proximity-ligation methods.\\n\")\n",
    "cat(paste0(\"Precisions were estimated to be equal to: \", \n",
    "           round(100*nrow(filter(CRISPRi_pos, MChIPC.loop))/nrow(filter(CRISPRi, MChIPC.loop)),1),\"% (MChIP-C),\",\n",
    "           round(100*nrow(filter(CRISPRi_pos, PLACseq.loop))/nrow(filter(CRISPRi, PLACseq.loop)),1),\"% (PLAC-seq),\",\n",
    "           round(100*nrow(filter(CRISPRi_pos, MicroC.loop))/nrow(filter(CRISPRi, MicroC.loop)),1),\"% (Micro-C) and \",\n",
    "           round(100*nrow(filter(CRISPRi_pos, HiC.loop))/nrow(filter(CRISPRi, HiC.loop)),1),\"% (Hi-C).\\n\"))\n",
    "cat(paste0(\"Finally we analyzed 50%-, 25%- and 10%-subsampled MChIP-C datasets. Their sensitivity was (correspondingly): \",\n",
    "          round(100*nrow(filter(CRISPRi_pos,MChIPC.sub05.loop))/nrow(CRISPRi_pos),1), \"%, \",\n",
    "          round(100*nrow(filter(CRISPRi_pos,MChIPC.sub025.loop))/nrow(CRISPRi_pos),1), \"%, and \",\n",
    "          round(100*nrow(filter(CRISPRi_pos,MChIPC.sub01.loop))/nrow(CRISPRi_pos),1), \"%. False positive rate: \",\n",
    "          round(100*nrow(filter(CRISPRi_neg,MChIPC.sub05.loop))/nrow(CRISPRi_neg),1), \"%, \",\n",
    "          round(100*nrow(filter(CRISPRi_neg,MChIPC.sub025.loop))/nrow(CRISPRi_neg),1), \"%, and \",\n",
    "          round(100*nrow(filter(CRISPRi_neg,MChIPC.sub01.loop))/nrow(CRISPRi_neg),1), \"%. Precision: \",\n",
    "          round(100*nrow(filter(CRISPRi_pos, MChIPC.sub05.loop))/nrow(filter(CRISPRi, MChIPC.sub05.loop)),1), \"%, \",\n",
    "          round(100*nrow(filter(CRISPRi_pos, MChIPC.sub025.loop))/nrow(filter(CRISPRi, MChIPC.sub025.loop)),1), \"%, and \",\n",
    "          round(100*nrow(filter(CRISPRi_pos, MChIPC.sub01.loop))/nrow(filter(CRISPRi, MChIPC.sub01.loop)),1), \"%.\"))               \n",
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "write.table(CRISPRi_pos, \"Summary_output_datasets/E-P_pairs.txt\", col.names=T, sep = \"\\t\")\n",
    "write.table(CRISPRi_neg, \"Summary_output_datasets/DHS-P_pairs.txt\", col.names=T, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7f846e5",
   "metadata": {},
   "source": [
    "# Plotting E(DHS)-P interaction heatmaps and average profiles"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8de4dd3a",
   "metadata": {},
   "source": [
    "# E-P pairs physically linked in MChIP-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dada5130",
   "metadata": {},
   "outputs": [],
   "source": [
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "MChIPC_signal <- read.csv(\"MChIPC_output/MChIPC_interaction_data.txt\", header = T, sep = \"\\t\")\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank!=0)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "MChIPC_signal <- MChIPC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "CRISPRi_pos_ovrl <- filter(CRISPRi_pos, MChIPC.loop==TRUE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_pos_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_ovrl))){\n",
    "  loop <- CRISPRi_pos_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MChIPC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(21,28)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_ovrl_heatmap <- left_join(CRISPRi_pos_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_pos_ovrl_heatmap[is.na(CRISPRi_pos_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_ovrl_heatmap)[2:ncol(CRISPRi_pos_ovrl_heatmap),])\n",
    "CRISPRi_pos_ovrl_heatmap <- arrange(CRISPRi_pos_ovrl_heatmap, rowMeans(CRISPRi_pos_ovrl_heatmap))\n",
    "CRISPRi_pos_ovrl_heatmap_mod <- CRISPRi_pos_ovrl_heatmap\n",
    "CRISPRi_pos_ovrl_heatmap_mod[CRISPRi_pos_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.4/4a_int_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70341c2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"MChIP-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ccd791a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.4/4a_int_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78443239",
   "metadata": {},
   "source": [
    "# E-P pairs with no physical link in MChIP-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9dd17c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "MChIPC_signal <- read.csv(\"MChIPC_output/MChIPC_interaction_data.txt\", header = T, sep = \"\\t\")\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank!=0)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "MChIPC_signal <- MChIPC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "CRISPRi_pos_non_ovrl <- filter(CRISPRi_pos, MChIPC.loop==FALSE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_pos_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_non_ovrl))){\n",
    "  loop <- CRISPRi_pos_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MChIPC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(21,28)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_non_ovrl_heatmap <- left_join(CRISPRi_pos_non_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_pos_non_ovrl_heatmap[is.na(CRISPRi_pos_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_non_ovrl_heatmap)[2:ncol(CRISPRi_pos_non_ovrl_heatmap),])\n",
    "CRISPRi_pos_non_ovrl_heatmap <- arrange(CRISPRi_pos_non_ovrl_heatmap, rowMeans(CRISPRi_pos_non_ovrl_heatmap))\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod <- CRISPRi_pos_non_ovrl_heatmap\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod[CRISPRi_pos_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.4/4a_none_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88b51c22",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"MChIP-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da725f9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.4/4a_none_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ae55e66",
   "metadata": {},
   "source": [
    "# DHS-P pairs physically linked in MChIP-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fbecc41c",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "MChIPC_signal <- read.csv(\"MChIPC_output/MChIPC_interaction_data.txt\", header = T, sep = \"\\t\")\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank!=0)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "MChIPC_signal <- MChIPC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "CRISPRi_neg_ovrl <- filter(CRISPRi_neg, MChIPC.loop==TRUE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_neg_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_ovrl))){\n",
    "  loop <- CRISPRi_neg_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MChIPC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(21,28)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_ovrl_heatmap <- left_join(CRISPRi_neg_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_neg_ovrl_heatmap[is.na(CRISPRi_neg_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_ovrl_heatmap)[2:ncol(CRISPRi_neg_ovrl_heatmap),])\n",
    "CRISPRi_neg_ovrl_heatmap <- arrange(CRISPRi_neg_ovrl_heatmap, rowMeans(CRISPRi_neg_ovrl_heatmap))\n",
    "CRISPRi_neg_ovrl_heatmap_mod <- CRISPRi_neg_ovrl_heatmap\n",
    "CRISPRi_neg_ovrl_heatmap_mod[CRISPRi_neg_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.4/4a_int_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f18d98f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from DHS (250bp bins)\")+ylab(\"MChIP-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "09485eb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.4/4a_int_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68bae93d",
   "metadata": {},
   "source": [
    "# DHS-P pairs with no physical link in MChIPC dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ccb511c",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "MChIPC_signal <- read.csv(\"MChIPC_output/MChIPC_interaction_data.txt\", header = T, sep = \"\\t\")\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank!=0)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "MChIPC_signal <- MChIPC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MChIPC_signal <- filter(MChIPC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MChIPC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MChIPC_signal$vp_start, end=MChIPC_signal$vp_end))\n",
    "CRISPRi_neg_non_ovrl <- filter(CRISPRi_neg, MChIPC.loop==FALSE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_neg_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_non_ovrl))){\n",
    "  loop <- CRISPRi_neg_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MChIPC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(21,28)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_non_ovrl_heatmap <- left_join(CRISPRi_neg_non_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_neg_non_ovrl_heatmap[is.na(CRISPRi_neg_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_non_ovrl_heatmap)[2:ncol(CRISPRi_neg_non_ovrl_heatmap),])\n",
    "CRISPRi_neg_non_ovrl_heatmap <- arrange(CRISPRi_neg_non_ovrl_heatmap, rowMeans(CRISPRi_neg_non_ovrl_heatmap))\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod <- CRISPRi_neg_non_ovrl_heatmap\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod[CRISPRi_neg_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.4/4a_none_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e32afff8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from DHS (250bp bins)\")+ylab(\"MChIP-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1191a8d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.4/4a_none_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9365636e",
   "metadata": {},
   "source": [
    "# E-P pairs physically linked in PLAC-seq dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfbe78a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "PLACseq_signal <- read.csv(\"PLACseq_output/PLACseq.signal.df.txt\", header = F, sep = \"\\t\")\n",
    "PLACseq_signal$V7 <- as.numeric(gsub(\",\",\".\",PLACseq_signal$V7))\n",
    "colnames(PLACseq_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\")\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "PLACseq_signal <- PLACseq_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "PLACseq_signal$dist_rank <- 0\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_start>PLACseq_signal$OE_start]<-(PLACseq_signal$vp_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$OE_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_end<PLACseq_signal$OE_start]<-(PLACseq_signal$OE_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$vp_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal <- filter(PLACseq_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "CRISPRi_pos_ovrl <- filter(CRISPRi_pos, PLACseq.loop==TRUE & overlaps.CTCF==FALSE &\n",
    "                          abs((target_site.start+target_site.stop)/2-target_gene.start)>10000)\n",
    "CRISPRi_pos_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_ovrl))){\n",
    "  loop <- CRISPRi_pos_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- PLACseq_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(8,7)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_ovrl_heatmap <- left_join(CRISPRi_pos_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_pos_ovrl_heatmap[is.na(CRISPRi_pos_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_ovrl_heatmap)[2:ncol(CRISPRi_pos_ovrl_heatmap),])\n",
    "CRISPRi_pos_ovrl_heatmap <- arrange(CRISPRi_pos_ovrl_heatmap, rowMeans(CRISPRi_pos_ovrl_heatmap))\n",
    "CRISPRi_pos_ovrl_heatmap_mod <- CRISPRi_pos_ovrl_heatmap\n",
    "CRISPRi_pos_ovrl_heatmap_mod[CRISPRi_pos_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_PLACseq_int_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd9f7b4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"PLAC-seq signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61098230",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_PLACseq_int_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb695e71",
   "metadata": {},
   "source": [
    "# E-P pairs with no physical link in PLAC-seq dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8ca1efc",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "PLACseq_signal <- read.csv(\"PLACseq_output/PLACseq.signal.df.txt\", header = F, sep = \"\\t\")\n",
    "PLACseq_signal$V7 <- as.numeric(gsub(\",\",\".\",PLACseq_signal$V7))\n",
    "colnames(PLACseq_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\")\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "PLACseq_signal <- PLACseq_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "PLACseq_signal$dist_rank <- 0\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_start>PLACseq_signal$OE_start]<-(PLACseq_signal$vp_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$OE_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_end<PLACseq_signal$OE_start]<-(PLACseq_signal$OE_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$vp_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal <- filter(PLACseq_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "CRISPRi_pos_non_ovrl <- filter(CRISPRi_pos, PLACseq.loop==FALSE & overlaps.CTCF==FALSE &\n",
    "                              abs((target_site.start+target_site.stop)/2-target_gene.start)>10000)\n",
    "CRISPRi_pos_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_non_ovrl))){\n",
    "  loop <- CRISPRi_pos_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- PLACseq_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(8,7)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_non_ovrl_heatmap <- left_join(CRISPRi_pos_non_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_pos_non_ovrl_heatmap[is.na(CRISPRi_pos_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_non_ovrl_heatmap)[2:ncol(CRISPRi_pos_non_ovrl_heatmap),])\n",
    "CRISPRi_pos_non_ovrl_heatmap <- arrange(CRISPRi_pos_non_ovrl_heatmap, rowMeans(CRISPRi_pos_non_ovrl_heatmap))\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod <- CRISPRi_pos_non_ovrl_heatmap\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod[CRISPRi_pos_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_PLACseq_none_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d936561",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"PLAC-seq signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "afa3cb29",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_PLACseq_none_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10a3c79b",
   "metadata": {},
   "source": [
    "# DHS-P pairs physically linked in PLAC-seq dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "026845f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "PLACseq_signal <- read.csv(\"PLACseq_output/PLACseq.signal.df.txt\", header = F, sep = \"\\t\")\n",
    "PLACseq_signal$V7 <- as.numeric(gsub(\",\",\".\",PLACseq_signal$V7))\n",
    "colnames(PLACseq_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\")\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "PLACseq_signal <- PLACseq_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "PLACseq_signal$dist_rank <- 0\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_start>PLACseq_signal$OE_start]<-(PLACseq_signal$vp_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$OE_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_end<PLACseq_signal$OE_start]<-(PLACseq_signal$OE_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$vp_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal <- filter(PLACseq_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "CRISPRi_neg_ovrl <- filter(CRISPRi_neg, PLACseq.loop==TRUE & overlaps.CTCF==FALSE &\n",
    "                          abs((target_site.start+target_site.stop)/2-target_gene.start)>10000)\n",
    "CRISPRi_neg_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_ovrl))){\n",
    "  loop <- CRISPRi_neg_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- PLACseq_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(8,7)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_ovrl_heatmap <- left_join(CRISPRi_neg_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_neg_ovrl_heatmap[is.na(CRISPRi_neg_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_ovrl_heatmap)[2:ncol(CRISPRi_neg_ovrl_heatmap),])\n",
    "CRISPRi_neg_ovrl_heatmap <- arrange(CRISPRi_neg_ovrl_heatmap, rowMeans(CRISPRi_neg_ovrl_heatmap))\n",
    "CRISPRi_neg_ovrl_heatmap_mod <- CRISPRi_neg_ovrl_heatmap\n",
    "CRISPRi_neg_ovrl_heatmap_mod[CRISPRi_neg_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_PLACseq_int_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db9c92b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from DHS (250bp bins)\")+ylab(\"PLAC-seq signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9595dbc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_PLACseq_int_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5323613",
   "metadata": {},
   "source": [
    "# DHS-P pairs with no physical link in PLAC-seq dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c291bdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "PLACseq_signal <- read.csv(\"PLACseq_output/PLACseq.signal.df.txt\", header = F, sep = \"\\t\")\n",
    "PLACseq_signal$V7 <- as.numeric(gsub(\",\",\".\",PLACseq_signal$V7))\n",
    "colnames(PLACseq_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\")\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "PLACseq_signal <- PLACseq_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "PLACseq_signal$dist_rank <- 0\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_start>PLACseq_signal$OE_start]<-(PLACseq_signal$vp_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$OE_start[PLACseq_signal$vp_start>PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal$dist_rank[PLACseq_signal$vp_end<PLACseq_signal$OE_start]<-(PLACseq_signal$OE_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start]- \n",
    "                                                            PLACseq_signal$vp_end[PLACseq_signal$vp_end<PLACseq_signal$OE_start])/250\n",
    "PLACseq_signal <- filter(PLACseq_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = PLACseq_signal$vp_chr, \n",
    "                         ranges=IRanges(start=PLACseq_signal$vp_start, end=PLACseq_signal$vp_end))\n",
    "CRISPRi_neg_non_ovrl <- filter(CRISPRi_neg, PLACseq.loop==FALSE & overlaps.CTCF==FALSE &\n",
    "                              abs((target_site.start+target_site.stop)/2-target_gene.start)>10000)\n",
    "CRISPRi_neg_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_non_ovrl))){\n",
    "  loop <- CRISPRi_neg_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- PLACseq_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal <- loop_signal[,c(8,7)]\n",
    "  colnames(loop_signal)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_non_ovrl_heatmap <- left_join(CRISPRi_neg_non_ovrl_heatmap, loop_signal)})}\n",
    "CRISPRi_neg_non_ovrl_heatmap[is.na(CRISPRi_neg_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_non_ovrl_heatmap)[2:ncol(CRISPRi_neg_non_ovrl_heatmap),])\n",
    "CRISPRi_neg_non_ovrl_heatmap <- arrange(CRISPRi_neg_non_ovrl_heatmap, rowMeans(CRISPRi_neg_non_ovrl_heatmap))\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod <- CRISPRi_neg_non_ovrl_heatmap\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod[CRISPRi_neg_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_PLACseq_none_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59f890fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from DHS (250bp bins)\")+ylab(\"PLAC-seq signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6aaf6e49",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_PLACseq_none_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce179686",
   "metadata": {},
   "source": [
    "# E-P pairs physically linked in Micro-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef343ace",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "MicroC_signal <- read.csv(\"MicroC_output/MicroC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(MicroC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "MicroC_signal <- MicroC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MicroC_signal$dist_rank <- 0\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_start>MicroC_signal$OE_start]<-(MicroC_signal$vp_start[MicroC_signal$vp_start>MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$OE_start[MicroC_signal$vp_start>MicroC_signal$OE_start])/250\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_end<MicroC_signal$OE_start]<-(MicroC_signal$OE_end[MicroC_signal$vp_end<MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$vp_end[MicroC_signal$vp_end<MicroC_signal$OE_start])/250\n",
    "MicroC_signal <- filter(MicroC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "CRISPRi_pos_ovrl <- filter(CRISPRi_pos, MicroC.loop==TRUE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_pos_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_pos_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_ovrl))){\n",
    "  loop <- CRISPRi_pos_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MicroC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_ovrl_heatmap <- left_join(CRISPRi_pos_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_pos_ovrl_heatmap_ICE <- left_join(CRISPRi_pos_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_pos_ovrl_heatmap[is.na(CRISPRi_pos_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_ovrl_heatmap_ICE[is.na(CRISPRi_pos_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_pos_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_ovrl_heatmap)[2:ncol(CRISPRi_pos_ovrl_heatmap),])\n",
    "CRISPRi_pos_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_pos_ovrl_heatmap_ICE)[2:ncol(CRISPRi_pos_ovrl_heatmap_ICE),])\n",
    "CRISPRi_pos_ovrl_heatmap <- arrange(CRISPRi_pos_ovrl_heatmap, rowMeans(CRISPRi_pos_ovrl_heatmap))\n",
    "CRISPRi_pos_ovrl_heatmap_ICE <- arrange(CRISPRi_pos_ovrl_heatmap_ICE, rowMeans(CRISPRi_pos_ovrl_heatmap_ICE))\n",
    "CRISPRi_pos_ovrl_heatmap_mod <- CRISPRi_pos_ovrl_heatmap\n",
    "CRISPRi_pos_ovrl_heatmap_mod[CRISPRi_pos_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_MicroC_int_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2814af42",
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93ab1a65",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_int_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3da8e2e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=1000*colMeans(CRISPRi_pos_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b188bee",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_int_EP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c2ff13a7",
   "metadata": {},
   "source": [
    "# E-P pairs with no physical link in Micro-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e3e55e10",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "MicroC_signal <- read.csv(\"MicroC_output/MicroC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(MicroC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "MicroC_signal <- MicroC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MicroC_signal$dist_rank <- 0\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_start>MicroC_signal$OE_start]<-(MicroC_signal$vp_start[MicroC_signal$vp_start>MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$OE_start[MicroC_signal$vp_start>MicroC_signal$OE_start])/250\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_end<MicroC_signal$OE_start]<-(MicroC_signal$OE_end[MicroC_signal$vp_end<MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$vp_end[MicroC_signal$vp_end<MicroC_signal$OE_start])/250\n",
    "MicroC_signal <- filter(MicroC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "CRISPRi_pos_non_ovrl <- filter(CRISPRi_pos, MicroC.loop==FALSE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_pos_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_non_ovrl))){\n",
    "  loop <- CRISPRi_pos_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MicroC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_non_ovrl_heatmap <- left_join(CRISPRi_pos_non_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_pos_non_ovrl_heatmap_ICE <- left_join(CRISPRi_pos_non_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_pos_non_ovrl_heatmap[is.na(CRISPRi_pos_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE[is.na(CRISPRi_pos_non_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_pos_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_non_ovrl_heatmap)[2:ncol(CRISPRi_pos_non_ovrl_heatmap),])\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_pos_non_ovrl_heatmap_ICE)[2:ncol(CRISPRi_pos_non_ovrl_heatmap_ICE),])\n",
    "CRISPRi_pos_non_ovrl_heatmap <- arrange(CRISPRi_pos_non_ovrl_heatmap, rowMeans(CRISPRi_pos_non_ovrl_heatmap))\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE <- arrange(CRISPRi_pos_non_ovrl_heatmap_ICE, rowMeans(CRISPRi_pos_non_ovrl_heatmap_ICE))\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod <- CRISPRi_pos_non_ovrl_heatmap\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod[CRISPRi_pos_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_MicroC_none_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9dd3abdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fdf7872c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_none_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2ffe18c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=1000*colMeans(CRISPRi_pos_non_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17e7d5e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_none_EP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "050a8b03",
   "metadata": {},
   "source": [
    "# DHS-P pairs physically linked in Micro-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b425145",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "MicroC_signal <- read.csv(\"MicroC_output/MicroC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(MicroC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "MicroC_signal <- MicroC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MicroC_signal$dist_rank <- 0\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_start>MicroC_signal$OE_start]<-(MicroC_signal$vp_start[MicroC_signal$vp_start>MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$OE_start[MicroC_signal$vp_start>MicroC_signal$OE_start])/250\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_end<MicroC_signal$OE_start]<-(MicroC_signal$OE_end[MicroC_signal$vp_end<MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$vp_end[MicroC_signal$vp_end<MicroC_signal$OE_start])/250\n",
    "MicroC_signal <- filter(MicroC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "CRISPRi_neg_ovrl <- filter(CRISPRi_neg, MicroC.loop==TRUE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_neg_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_neg_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_ovrl))){\n",
    "  loop <- CRISPRi_neg_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MicroC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_ovrl_heatmap <- left_join(CRISPRi_neg_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_neg_ovrl_heatmap_ICE <- left_join(CRISPRi_neg_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_neg_ovrl_heatmap[is.na(CRISPRi_neg_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_ovrl_heatmap_ICE[is.na(CRISPRi_neg_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_neg_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_ovrl_heatmap)[2:ncol(CRISPRi_neg_ovrl_heatmap),])\n",
    "CRISPRi_neg_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_neg_ovrl_heatmap_ICE)[2:ncol(CRISPRi_neg_ovrl_heatmap_ICE),])\n",
    "CRISPRi_neg_ovrl_heatmap <- arrange(CRISPRi_neg_ovrl_heatmap, rowMeans(CRISPRi_neg_ovrl_heatmap))\n",
    "CRISPRi_neg_ovrl_heatmap_ICE <- arrange(CRISPRi_neg_ovrl_heatmap_ICE, rowMeans(CRISPRi_neg_ovrl_heatmap_ICE))\n",
    "CRISPRi_neg_ovrl_heatmap_mod <- CRISPRi_neg_ovrl_heatmap\n",
    "CRISPRi_neg_ovrl_heatmap_mod[CRISPRi_neg_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_MicroC_int_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e11e24b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d128829",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_int_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da9cd306",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=1000*colMeans(CRISPRi_neg_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a6766ee5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_int_DHSP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6a3f982",
   "metadata": {},
   "source": [
    "# DHS-P pairs with no physical link in Micro-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f557853",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "MicroC_signal <- read.csv(\"MicroC_output/MicroC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(MicroC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "MicroC_signal <- MicroC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "MicroC_signal$dist_rank <- 0\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_start>MicroC_signal$OE_start]<-(MicroC_signal$vp_start[MicroC_signal$vp_start>MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$OE_start[MicroC_signal$vp_start>MicroC_signal$OE_start])/250\n",
    "MicroC_signal$dist_rank[MicroC_signal$vp_end<MicroC_signal$OE_start]<-(MicroC_signal$OE_end[MicroC_signal$vp_end<MicroC_signal$OE_start]- \n",
    "                                                            MicroC_signal$vp_end[MicroC_signal$vp_end<MicroC_signal$OE_start])/250\n",
    "MicroC_signal <- filter(MicroC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = MicroC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=MicroC_signal$vp_start, end=MicroC_signal$vp_end))\n",
    "CRISPRi_neg_non_ovrl <- filter(CRISPRi_neg, MicroC.loop==FALSE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_neg_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_non_ovrl))){\n",
    "  loop <- CRISPRi_neg_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- MicroC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_non_ovrl_heatmap <- left_join(CRISPRi_neg_non_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_neg_non_ovrl_heatmap_ICE <- left_join(CRISPRi_neg_non_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_neg_non_ovrl_heatmap[is.na(CRISPRi_neg_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE[is.na(CRISPRi_neg_non_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_neg_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_non_ovrl_heatmap)[2:ncol(CRISPRi_neg_non_ovrl_heatmap),])\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_neg_non_ovrl_heatmap_ICE)[2:ncol(CRISPRi_neg_non_ovrl_heatmap_ICE),])\n",
    "CRISPRi_neg_non_ovrl_heatmap <- arrange(CRISPRi_neg_non_ovrl_heatmap, rowMeans(CRISPRi_neg_non_ovrl_heatmap))\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE <- arrange(CRISPRi_neg_non_ovrl_heatmap_ICE, rowMeans(CRISPRi_neg_non_ovrl_heatmap_ICE))\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod <- CRISPRi_neg_non_ovrl_heatmap\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod[CRISPRi_neg_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_MicroC_none_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2abcd0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7709a37",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_none_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb91f9ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=1000*colMeans(CRISPRi_neg_non_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Micro-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ffe0cc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_MicroC_none_DHSP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0c72418",
   "metadata": {},
   "source": [
    "# E-P pairs physically linked in Hi-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0487b02",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "HiC_signal <- read.csv(\"HiC_output/HiC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(HiC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "HiC_signal <- HiC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "HiC_signal$dist_rank <- 0\n",
    "HiC_signal$dist_rank[HiC_signal$vp_start>HiC_signal$OE_start]<-(HiC_signal$vp_start[HiC_signal$vp_start>HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$OE_start[HiC_signal$vp_start>HiC_signal$OE_start])/250\n",
    "HiC_signal$dist_rank[HiC_signal$vp_end<HiC_signal$OE_start]<-(HiC_signal$OE_end[HiC_signal$vp_end<HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$vp_end[HiC_signal$vp_end<HiC_signal$OE_start])/250\n",
    "HiC_signal <- filter(HiC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "CRISPRi_pos_ovrl <- filter(CRISPRi_pos, HiC.loop==TRUE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_pos_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_pos_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_ovrl))){\n",
    "  loop <- CRISPRi_pos_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- HiC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_ovrl_heatmap <- left_join(CRISPRi_pos_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_pos_ovrl_heatmap_ICE <- left_join(CRISPRi_pos_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_pos_ovrl_heatmap[is.na(CRISPRi_pos_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_ovrl_heatmap_ICE[is.na(CRISPRi_pos_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_pos_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_ovrl_heatmap)[2:ncol(CRISPRi_pos_ovrl_heatmap),])\n",
    "CRISPRi_pos_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_pos_ovrl_heatmap_ICE)[2:ncol(CRISPRi_pos_ovrl_heatmap_ICE),])\n",
    "CRISPRi_pos_ovrl_heatmap <- arrange(CRISPRi_pos_ovrl_heatmap, rowMeans(CRISPRi_pos_ovrl_heatmap))\n",
    "CRISPRi_pos_ovrl_heatmap_ICE <- arrange(CRISPRi_pos_ovrl_heatmap_ICE, rowMeans(CRISPRi_pos_ovrl_heatmap_ICE))\n",
    "CRISPRi_pos_ovrl_heatmap_mod <- CRISPRi_pos_ovrl_heatmap\n",
    "CRISPRi_pos_ovrl_heatmap_mod[CRISPRi_pos_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_HiC_int_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f182f209",
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a117c74",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_int_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38784beb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80bbe985",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_int_EP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25da7d92",
   "metadata": {},
   "source": [
    "# E-P pairs with no physical link in Hi-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bbacc90",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_pos$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_pos$target_gene.start, end=CRISPRi_pos$target_gene.start+1))\n",
    "HiC_signal <- read.csv(\"HiC_output/HiC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(HiC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "HiC_signal <- HiC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "HiC_signal$dist_rank <- 0\n",
    "HiC_signal$dist_rank[HiC_signal$vp_start>HiC_signal$OE_start]<-(HiC_signal$vp_start[HiC_signal$vp_start>HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$OE_start[HiC_signal$vp_start>HiC_signal$OE_start])/250\n",
    "HiC_signal$dist_rank[HiC_signal$vp_end<HiC_signal$OE_start]<-(HiC_signal$OE_end[HiC_signal$vp_end<HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$vp_end[HiC_signal$vp_end<HiC_signal$OE_start])/250\n",
    "HiC_signal <- filter(HiC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "CRISPRi_pos_non_ovrl <- filter(CRISPRi_pos, HiC.loop==FALSE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_pos_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_pos_non_ovrl))){\n",
    "  loop <- CRISPRi_pos_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- HiC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_pos_non_ovrl_heatmap <- left_join(CRISPRi_pos_non_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_pos_non_ovrl_heatmap_ICE <- left_join(CRISPRi_pos_non_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_pos_non_ovrl_heatmap[is.na(CRISPRi_pos_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE[is.na(CRISPRi_pos_non_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_pos_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_pos_non_ovrl_heatmap)[2:ncol(CRISPRi_pos_non_ovrl_heatmap),])\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_pos_non_ovrl_heatmap_ICE)[2:ncol(CRISPRi_pos_non_ovrl_heatmap_ICE),])\n",
    "CRISPRi_pos_non_ovrl_heatmap <- arrange(CRISPRi_pos_non_ovrl_heatmap, rowMeans(CRISPRi_pos_non_ovrl_heatmap))\n",
    "CRISPRi_pos_non_ovrl_heatmap_ICE <- arrange(CRISPRi_pos_non_ovrl_heatmap_ICE, rowMeans(CRISPRi_pos_non_ovrl_heatmap_ICE))\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod <- CRISPRi_pos_non_ovrl_heatmap\n",
    "CRISPRi_pos_non_ovrl_heatmap_mod[CRISPRi_pos_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_HiC_none_EP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_pos_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c2fa316",
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "373a9b64",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_none_EP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8936907",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_pos_non_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4d8080f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_none_EP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cacaa340",
   "metadata": {},
   "source": [
    "# DHS-P pairs physically linked in Hi-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1400eb03",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "HiC_signal <- read.csv(\"HiC_output/HiC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(HiC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "HiC_signal <- HiC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "HiC_signal$dist_rank <- 0\n",
    "HiC_signal$dist_rank[HiC_signal$vp_start>HiC_signal$OE_start]<-(HiC_signal$vp_start[HiC_signal$vp_start>HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$OE_start[HiC_signal$vp_start>HiC_signal$OE_start])/250\n",
    "HiC_signal$dist_rank[HiC_signal$vp_end<HiC_signal$OE_start]<-(HiC_signal$OE_end[HiC_signal$vp_end<HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$vp_end[HiC_signal$vp_end<HiC_signal$OE_start])/250\n",
    "HiC_signal <- filter(HiC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "CRISPRi_neg_ovrl <- filter(CRISPRi_neg, HiC.loop==TRUE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_neg_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_neg_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_ovrl))){\n",
    "  loop <- CRISPRi_neg_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- HiC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_ovrl_heatmap <- left_join(CRISPRi_neg_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_neg_ovrl_heatmap_ICE <- left_join(CRISPRi_neg_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_neg_ovrl_heatmap[is.na(CRISPRi_neg_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_ovrl_heatmap_ICE[is.na(CRISPRi_neg_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_neg_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_ovrl_heatmap)[2:ncol(CRISPRi_neg_ovrl_heatmap),])\n",
    "CRISPRi_neg_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_neg_ovrl_heatmap_ICE)[2:ncol(CRISPRi_neg_ovrl_heatmap_ICE),])\n",
    "CRISPRi_neg_ovrl_heatmap <- arrange(CRISPRi_neg_ovrl_heatmap, rowMeans(CRISPRi_neg_ovrl_heatmap))\n",
    "CRISPRi_neg_ovrl_heatmap_ICE <- arrange(CRISPRi_neg_ovrl_heatmap_ICE, rowMeans(CRISPRi_neg_ovrl_heatmap_ICE))\n",
    "CRISPRi_neg_ovrl_heatmap_mod <- CRISPRi_neg_ovrl_heatmap\n",
    "CRISPRi_neg_ovrl_heatmap_mod[CRISPRi_neg_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_HiC_int_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa45e794",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd53b2b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_int_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f085300",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22cac87a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_int_DHSP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba6c00c6",
   "metadata": {},
   "source": [
    "# DHS-P pairs with no physical link in Hi-C dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1117c379",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=setdiff(ls(), c(\"CRISPRi_pos\", \"CRISPRi_neg\")))\n",
    "# heatmap\n",
    "CRISPRi_ranges <- GRanges(seqnames = CRISPRi_neg$target_site.chr, \n",
    "                          ranges=IRanges(start=CRISPRi_neg$target_gene.start, end=CRISPRi_neg$target_gene.start+1))\n",
    "HiC_signal <- read.csv(\"HiC_output/HiC.signal.df.txt\", header = F, sep = \"\\t\") %>% filter(V1!=\"\")\n",
    "colnames(HiC_signal) <- c(\"vp_chr\",\"vp_start\",\"vp_end\",\"OE_chr\",\"OE_start\",\"OE_end\",\"N_sum\", \"N_sum_ICE\")\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "HiC_signal <- HiC_signal[unique(as.data.frame(findOverlaps(signal_ranges, CRISPRi_ranges))$queryHits),]\n",
    "HiC_signal$dist_rank <- 0\n",
    "HiC_signal$dist_rank[HiC_signal$vp_start>HiC_signal$OE_start]<-(HiC_signal$vp_start[HiC_signal$vp_start>HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$OE_start[HiC_signal$vp_start>HiC_signal$OE_start])/250\n",
    "HiC_signal$dist_rank[HiC_signal$vp_end<HiC_signal$OE_start]<-(HiC_signal$OE_end[HiC_signal$vp_end<HiC_signal$OE_start]- \n",
    "                                                            HiC_signal$vp_end[HiC_signal$vp_end<HiC_signal$OE_start])/250\n",
    "HiC_signal <- filter(HiC_signal, dist_rank>19)\n",
    "signal_ranges <- GRanges(seqnames = HiC_signal$vp_chr, \n",
    "                         ranges=IRanges(start=HiC_signal$vp_start, end=HiC_signal$vp_end))\n",
    "CRISPRi_neg_non_ovrl <- filter(CRISPRi_neg, HiC.loop==FALSE & overlaps.CTCF==FALSE)\n",
    "CRISPRi_neg_non_ovrl_heatmap <- tibble(dist_rank=seq(-19,19))\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE <- tibble(dist_rank=seq(-19,19))\n",
    "for (i in seq(nrow(CRISPRi_neg_non_ovrl))){\n",
    "  loop <- CRISPRi_neg_non_ovrl[i,]\n",
    "  loop_ranges <- GRanges(seqnames = loop$target_site.chr, \n",
    "                         ranges=IRanges(start=loop$target_gene.start, end=loop$target_gene.start+1))\n",
    "  loop_signal <- HiC_signal[as.data.frame(findOverlaps(signal_ranges, loop_ranges))$queryHits,]\n",
    "  loop_signal[loop_signal$vp_start > loop_signal$OE_start,]$dist_rank <- -1 * loop_signal[loop_signal$vp_start > \n",
    "                                                                                loop_signal$OE_start,]$dist_rank\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal <- filter(loop_signal, dist_rank >\n",
    "                            floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)-20 &\n",
    "                            dist_rank<floor(((loop$target_site.start+loop$target_site.stop)/2 - vp_start)/250)+20)}\n",
    "      else {loop_signal <- filter(loop_signal, dist_rank>ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)-20 & dist_rank<ceiling(((loop$target_site.start+\n",
    "            loop$target_site.stop)/2 - vp_end)/250)+20)}\n",
    "  if (loop$target_gene.start > loop$target_site.start) {loop_signal$dist_rank <- loop_signal$dist_rank -\n",
    "     floor(((loop$target_site.start+loop$target_site.stop)/2 - loop_signal$vp_start)/250)} \n",
    "    else {loop_signal$dist_rank <- loop_signal$dist_rank - ceiling(((loop$target_site.start+\n",
    "          loop$target_site.stop)/2 - loop_signal$vp_end)/250)}\n",
    "  loop_signal_n <- loop_signal[,c(9,7)]\n",
    "  loop_signal_ICE <- loop_signal[,c(9,8)]\n",
    "  colnames(loop_signal_n)[2] <- paste0(\"loop_\",i)\n",
    "  colnames(loop_signal_ICE)[2] <- paste0(\"loop_\",i)\n",
    "  suppressMessages({CRISPRi_neg_non_ovrl_heatmap <- left_join(CRISPRi_neg_non_ovrl_heatmap, loop_signal_n)})\n",
    "  suppressMessages({CRISPRi_neg_non_ovrl_heatmap_ICE <- left_join(CRISPRi_neg_non_ovrl_heatmap_ICE, loop_signal_ICE)})}\n",
    "CRISPRi_neg_non_ovrl_heatmap[is.na(CRISPRi_neg_non_ovrl_heatmap)] <- 0\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE[is.na(CRISPRi_neg_non_ovrl_heatmap_ICE)] <- 0\n",
    "CRISPRi_neg_non_ovrl_heatmap <- as.data.frame(t(CRISPRi_neg_non_ovrl_heatmap)[2:ncol(CRISPRi_neg_non_ovrl_heatmap),])\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE <- as.data.frame(t(CRISPRi_neg_non_ovrl_heatmap_ICE)[2:ncol(CRISPRi_neg_non_ovrl_heatmap_ICE),])\n",
    "CRISPRi_neg_non_ovrl_heatmap <- arrange(CRISPRi_neg_non_ovrl_heatmap, rowMeans(CRISPRi_neg_non_ovrl_heatmap))\n",
    "CRISPRi_neg_non_ovrl_heatmap_ICE <- arrange(CRISPRi_neg_non_ovrl_heatmap_ICE, rowMeans(CRISPRi_neg_non_ovrl_heatmap_ICE))\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod <- CRISPRi_neg_non_ovrl_heatmap\n",
    "CRISPRi_neg_non_ovrl_heatmap_mod[CRISPRi_neg_non_ovrl_heatmap_mod>20] <- 20\n",
    "pdf(\"Figures/Fig.S6/S6a_HiC_none_DHSP_heatmap.pdf\")\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)\n",
    "dev.off()\n",
    "heatmap(as.matrix(CRISPRi_neg_non_ovrl_heatmap_mod), Colv=NA, Rowv=NA, scale=\"none\", labRow=FALSE, labCol=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86529d34",
   "metadata": {},
   "outputs": [],
   "source": [
    "# raw profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_non_ovrl_heatmap)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31b5f893",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_none_DHSP_profile.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d1831a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ICE profile\n",
    "ggplot()+geom_line(aes(x=seq(-19,19,1),y=colMeans(CRISPRi_neg_non_ovrl_heatmap_ICE)),size=2)+theme_classic(base_size = 20)+\n",
    "xlim(-19,19) + ylim(0,17) + xlab(\"Distance from enhancer (250bp bins)\")+ylab(\"Hi-C signal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31d05b9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ggsave(\"Figures/Fig.S6/S6a_HiC_none_DHSP_profile_ICE.pdf\",device=\"pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a14afa1",
   "metadata": {},
   "source": [
    "# Plotting precision-recall and ROC curves"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36186492",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(list=ls())\n",
    "# loading E-P and DHS-P data and combining them\n",
    "positive <- read.csv(\"Summary_output_datasets/E-P_pairs.txt\", header = T, sep = \"\\t\")\n",
    "negative <- read.csv(\"Summary_output_datasets/DHS-P_pairs.txt\", header = T, sep = \"\\t\")\n",
    "combined <- rbind(positive, negative)\n",
    "all_TSS <- read.csv(\"Auxiliary_data/RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS.500bp.bed\", header = F, sep = \"\\t\")\n",
    "expression <- read.csv(\"Auxiliary_data/ENCFF934YBO.tsv\", header = T, sep = \"\\t\")[,c(1,6)]\n",
    "genes <- readGFF(\"Auxiliary_data/gencode.v19.annotation.gtf.gz\")\n",
    "genes <- filter(genes, type==\"gene\")[,c(9,13)] %>% distinct() \n",
    "expression <- left_join(expression,genes)\n",
    "all_TSS <- left_join(all_TSS, expression[,c(2,3)], by=c(\"V4\"=\"gene_name\")) %>% filter(TPM>0.5)\n",
    "output <- data.frame()\n",
    "chromosomes <- unique(all_TSS$V1)\n",
    "for (chromosome in chromosomes){\n",
    "  combined_ss <- filter(combined, target_site.chr==chromosome)\n",
    "  all_TSS_ss <- filter(all_TSS, V1==chromosome)\n",
    "  combined_ranges <- GRanges(seqnames = combined_ss$target_site.chr, ranges=IRanges(start=combined_ss$target_site.start, end=combined_ss$target_site.stop, names = seq(1,nrow(combined_ss))))\n",
    "  TSS_ranges <- GRanges(seqnames = all_TSS_ss$V1, ranges=IRanges(start=all_TSS_ss$V2, end=all_TSS_ss$V3,  names = all_TSS_ss$V4))\n",
    "  combined_ss$closest_gene_k1 <- all_TSS_ss$V4[transpose(as.data.frame(do.call(cbind, nearestKNeighbors(combined_ranges, TSS_ranges, k=1))))$V1]==combined_ss$gene_short_name \n",
    "  combined_ss$closest_gene_k2 <- ((all_TSS_ss$V4[transpose(as.data.frame(do.call(cbind, nearestKNeighbors(combined_ranges, TSS_ranges, k=2))))$V1]==combined_ss$gene_short_name) | (all_TSS_ss$V4[transpose(as.data.frame(do.call(cbind, nearestKNeighbors(combined_ranges, TSS_ranges, k=2))))$V2]==combined_ss$gene_short_name))\n",
    "  output <- bind_rows(output,combined_ss)\n",
    "}\n",
    "\n",
    "output$too_close_for_PLAC <- pmin(abs(output$target_gene.start - output$target_site.start), abs(output$target_gene.start - output$target_site.stop)) < 10000\n",
    "\n",
    "sensitivity <- c(nrow(filter(output, Significant&MChIPC.loop))/nrow(filter(output, Significant)), # sensitivity mchipc\n",
    "nrow(filter(output, Significant&MChIPC.sub05.loop))/nrow(filter(output, Significant)), # sensitivity 50% ss mchipc\n",
    "nrow(filter(output, Significant&MChIPC.sub025.loop))/nrow(filter(output, Significant)), # sensitivity 25% ss mchipc\n",
    "nrow(filter(output, Significant&MChIPC.sub01.loop))/nrow(filter(output, Significant)), # sensitivity 10% ss mchipc\n",
    "nrow(filter(output, Significant&PLAC.loop&!(to_close_for_PLAC)))/nrow(filter(output, Significant&!(to_close_for_PLAC))), # sensitivity plac-seq\n",
    "nrow(filter(output, Significant&MicroC.loop))/nrow(filter(output, Significant)), # sensitivity MicroC\n",
    "nrow(filter(output, Significant&HiC.loop))/nrow(filter(output, Significant)), # sensitivity HiC\n",
    "nrow(filter(output, Significant&closest_gene_k1))/nrow(filter(output, Significant)),# sensitivity the closest active promoter\n",
    "nrow(filter(output, Significant&closest_gene_k2))/nrow(filter(output, Significant)))# sensitivity the 2 closest active promoters\n",
    "\n",
    "precision <- c(nrow(filter(output, Significant&MChIPC.loop))/nrow(filter(output, MChIPC.loop)), # precision mchipc\n",
    "nrow(filter(output, Significant&MChIPC.sub05.loop))/nrow(filter(output, MChIPC.sub05.loop)), # precision 50% ss mchipc\n",
    "nrow(filter(output, Significant&MChIPC.sub025.loop))/nrow(filter(output, MChIPC.sub025.loop)), # precision 25% ss mchipc\n",
    "nrow(filter(output, Significant&MChIPC.sub01.loop))/nrow(filter(output, MChIPC.sub01.loop)), # precision 10% ss mchipc\n",
    "nrow(filter(output, Significant&PLAC.loop&!(to_close_for_PLAC)))/nrow(filter(output, PLAC.loop&!(to_close_for_PLAC))), # precision plac-seq\n",
    "nrow(filter(output, Significant&MicroC.loop))/nrow(filter(output, MicroC.loop)), # precision MicroC\n",
    "nrow(filter(output, Significant&HiC.loop))/nrow(filter(output, HiC.loop)), # precision HiC\n",
    "nrow(filter(output, Significant&closest_gene_k1))/nrow(filter(output, closest_gene_k1)), # precision the closest active promoter\n",
    "nrow(filter(output, Significant&closest_gene_k2))/nrow(filter(output, closest_gene_k2)))# precision the 2 closest active promoters\n",
    "\n",
    "FPR <- c(nrow(filter(output, !(Significant)&MChIPC.loop))/nrow(filter(output, !(Significant))), # FPR mchipc\n",
    "nrow(filter(output, !(Significant)&MChIPC.sub05.loop))/nrow(filter(output, !(Significant))), # FPR 50% ss mchipc\n",
    "nrow(filter(output, !(Significant)&MChIPC.sub025.loop))/nrow(filter(output, !(Significant))), # FPR 25% ss mchipc\n",
    "nrow(filter(output, !(Significant)&MChIPC.sub01.loop))/nrow(filter(output, !(Significant))), # FPR 10% ss mchipc\n",
    "nrow(filter(output, !(Significant)&PLAC.loop&!(to_close_for_PLAC)))/nrow(filter(output, !(Significant)&!(to_close_for_PLAC))), # FPR plac-seq\n",
    "nrow(filter(output, !(Significant)&MicroC.loop))/nrow(filter(output, !(Significant))), # FPR MicroC\n",
    "nrow(filter(output, !(Significant)&HiC.loop))/nrow(filter(output, !(Significant))), # FPR HiC\n",
    "nrow(filter(output, !(Significant)&closest_gene_k1))/nrow(filter(output, !(Significant))), # FPR the closest active promoter\n",
    "nrow(filter(output, !(Significant)&closest_gene_k2))/nrow(filter(output, !(Significant))))# FPR the 2 closest active promoters\n",
    "\n",
    "bm_data <- data.frame(sensitivity, precision, FPR) \n",
    "row.names(bm_data) <- c(\"MChIPC\", \"MChIPC.sub05\",\"MChIPC.sub025\", \"MChIPC.sub01\", \"PLAC\", \"MicroC\",\"HiC\", \"closest_k1\", \"closest_k2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32f23204",
   "metadata": {},
   "outputs": [],
   "source": [
    "pr_distance <- pr.curve(scores.class0=1/(abs(output$target_gene.start-output$target_site.start)), weights.class0=output$Significant, curve = TRUE)\n",
    "pdf(\"Figures/Fig.S6/S6b_PR_curve.pdf\")\n",
    "plot.new()\n",
    "grid(nx = NULL, ny = NULL, lty = 0.2, lwd = 1)\n",
    "par(new = TRUE)\n",
    "plot(pr_distance, color=1, auc.main = FALSE, main=\"\", cex.lab=1.5, cex.axis=1.5, ylim=c(0,0.3))\n",
    "points(bm_data$sensitivity, bm_data$precision, pch = 20)\n",
    "text(bm_data$sensitivity, bm_data$precision, labels=rownames(bm_data), adj=c(0.5,-0.5), cex=1)\n",
    "dev.off()\n",
    "plot.new()\n",
    "grid(nx = NULL, ny = NULL, lty = 0.2, lwd = 1)\n",
    "par(new = TRUE)\n",
    "plot(pr_distance, color=1, auc.main = FALSE, main=\"\", cex.lab=1.5, cex.axis=1.5, ylim=c(0,0.3))\n",
    "points(bm_data$sensitivity, bm_data$precision, pch = 20)\n",
    "text(bm_data$sensitivity, bm_data$precision, labels=rownames(bm_data), adj=c(0.5,-0.5), cex=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95d8b19c",
   "metadata": {},
   "outputs": [],
   "source": [
    "roc.distance <- roc.curve( scores.class0=1/(abs(output$target_gene.start-output$target_site.start)), weights.class0=output$Significant, curve = TRUE)\n",
    "pdf(\"Figures/Fig.S6/S6b_ROC_curve.pdf\")\n",
    "plot.new()\n",
    "grid(nx = NULL, ny = NULL, lty = 0.2, lwd = 1)\n",
    "par(new = TRUE)\n",
    "plot(roc.distance, color=1, auc.main = FALSE, main=\"\", cex.lab=1.5, cex.axis=1.5, xlim=c(0,0.3))\n",
    "points(bm_data$FPR, bm_data$sensitivity, pch = 20)\n",
    "text(bm_data$FPR, bm_data$sensitivity, labels=rownames(bm_data), adj=c(-0.2,0.5), cex=1)\n",
    "dev.off()\n",
    "plot.new()\n",
    "grid(nx = NULL, ny = NULL, lty = 0.2, lwd = 1)\n",
    "par(new = TRUE)\n",
    "plot(roc.distance, color=1, auc.main = FALSE, main=\"\", cex.lab=1.5, cex.axis=1.5, xlim=c(0,0.3))\n",
    "points(bm_data$FPR, bm_data$sensitivity, pch = 20)\n",
    "text(bm_data$FPR, bm_data$sensitivity, labels=rownames(bm_data), adj=c(-0.2,0.5), cex=1)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
